src/main/java/seedu/address/model/person/Major.java

```diff
@@ -0,0 +1,110 @@
+
+/**
+ * Represents a Student's name in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidMajor(String)}
+ */
+public class Major {
+
+    public static final String MESSAGE_CONSTRAINTS = "Majors should only contain alphabets, and is case sensitive!"
+            + "\n The valid computing majors are: CS (Computer Science), IS (Information Systems)"
+            + ", ISEC (Information Security), and BZA (Business Analytics)";
+
+    /*
+     * The first character of the address must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}]*";
+
+    /*
+     * The HashMap that maps the major, to the list of students that are currently
+     * in that major.
+     */
+    private static final HashMap<Major, ArrayList<Person>> studentsToMajors = new HashMap<>();
+
+    public final String value;
+
+    /**
+     * Constructs a {@code Name}.
+     *
+     * @param major A valid major.
+     */
+    public Major(String major) {
+        requireNonNull(major);
+        checkArgument(isValidMajor(major), MESSAGE_CONSTRAINTS);
+        value = major;
+    }
+
+    /**
+     * Adds a student to the array list corresponding to that student's major
+     *
+     * @param student The student to be added
+     */
+    public static void addStudent(Person student) {
+        Major majorOfStudent = student.getMajor();
+        Optional<ArrayList<Person>> currStudents = Optional.ofNullable(studentsToMajors.get(majorOfStudent));
+
+        if (currStudents.isEmpty()) {
+            ArrayList<Person> studentsInMajor = new ArrayList<>();
+            studentsInMajor.add(student);
+            studentsToMajors.put(majorOfStudent, studentsInMajor);
+        } else {
+            currStudents.get().add(student);
+            studentsToMajors.put(majorOfStudent, currStudents.get());
+        }
+    }
+
+    /**
+     * Remove student from the arraylist corresponding to the student's major
+     *
+     * @param student The student to be removed
+     */
+    public static void removeStudent(Person student) {
+        Major majorOfStudent = student.getMajor();
+        ArrayList<Person> currStudents = studentsToMajors.get(majorOfStudent);
+        currStudents.remove(student);
+    }
+
+    /**
+     * Returns true if a given string is a valid name.
+     */
+    public static boolean isValidMajor(String test) {
+        boolean matchesRegex = test.matches(VALIDATION_REGEX);
+        boolean validCourse = isValidMajorName(test);
+        return matchesRegex && validCourse;
+    }
+
+    /**
+     * Checks if the given major string is a valid major name
+     *
+     * @param major The major given
+     * @return True if valid, false if invalid
+     */
+    public static boolean isValidMajorName(String major) {
+        return major.equals("CS") || major.equals("IS") || major.equals("ISEC") || major.equals("BZA");
```

Comment:
Is the major field when creating a student case sensitive? If it isn't then just a suggestion that we could try making the majors non case-sensitive in the future